HIP/hip-external-transaction-signing.md

```diff
@@ -0,0 +1,337 @@
+- hip: &lt;HIP number (this is determined by the HIP editor)>
+- title: External Transaction Signing for SDK and other clients
+- author: Danno Ferrin (danno.ferrin@hedera.com)
+- type: Standards Track
+- category: Application
+- status: Draft
+- created: 19 Oct 2021
+- discussions-to: &lt;a URL pointing to the official discussion thread>
+- updated: &lt;comma separated list of dates>
+- requires: hip-caip-identifiers
+- replaces: &lt;HIP number(s)>
+- superseded-by: &lt;HIP number(s)>
+
+## Abstract
+
+An RPC protocol is described that would allow clients and SDKs to sign
+transactions with external programs, devices, and services.
+
+## Motivation
+
+The current Hedera SDK requires that the private keys for accounts be available
+and signed in process. One process is to provide the private key in an
+environmental variable and use local libraries to produces the signatures. This
+is a common approach used in Continuous Build systems and when properly handled
+provides good security. However, these keys can unlock large amounts of
+real-world value and the needed security posture needs to be better than "good."
+Typical high security setups involve hardware devices storing keys or external
+services with actual people gate-keeping activity. A process needs to be adopted
+that can seamlessly support both developer grade security and full production
+security.
+
+## Rationale
+
+As part of a larger DLT ecosystem Hedera should be better served by adopting
+existing cross-chain standards. Hence, the utility offered by teh Chain Agnostic
+Improvement Proposal group relating to RPC interactions was chosen as the main
+interface. In particular CAIP-25 and CAIP-27 provide the bulk of hte needed
+protocol. Some additional definitions of hedera signing operations is needed to
+finish our the full loop.
```

Comment:
"our the"? I think you want either just "the" or "our"

Reply:
just 'the' is what I went with.